hack/generator/pkg/astbuilder/assignments.go

```diff
@@ -6,26 +6,59 @@
 package astbuilder
 
 import (
+	"fmt"
 	"go/token"
 
 	"github.com/dave/dst"
 )
 
 // SimpleAssignment performs a simple assignment like:
-//     <lhs> := <rhs>       // tok = token.DEFINE
-// or  <lhs> = <rhs>        // tok = token.ASSIGN
-func SimpleAssignment(lhs dst.Expr, tok token.Token, rhs dst.Expr) *dst.AssignStmt {
+//
+//     <lhs> = <rhs>
+//
+func SimpleAssignment(lhs dst.Expr, rhs dst.Expr) *dst.AssignStmt {
 	return &dst.AssignStmt{
 		Lhs: []dst.Expr{
 			dst.Clone(lhs).(dst.Expr),
 		},
-		Tok: tok,
+		Tok: token.ASSIGN,
 		Rhs: []dst.Expr{
 			dst.Clone(rhs).(dst.Expr),
 		},
 	}
 }
 
+// SimpleDeclaration performs a simple assignment like:
+//
+//     <id> := <rhs>
+//
+func SimpleDeclaration(id string, rhs dst.Expr) *dst.AssignStmt {
+	return &dst.AssignStmt{
+		Lhs: []dst.Expr{
+			dst.NewIdent(id),
+		},
+		Tok: token.DEFINE,
+		Rhs: []dst.Expr{
+			dst.Clone(rhs).(dst.Expr),
+		},
+	}
+}
+
+// SetVariable allows for either variable declaration or assignment by passing the required token
+// Only token.DEFINE and token.ASSIGN are supported, other values will panic.
+// Use SimpleAssignment or SimpleDeclaration by preference
```

Comment:
```suggestion
// Use SimpleAssignment or SimpleDeclaration if possible
```
or something?

For whatever reason the "by preference" ending here makes it sound like: "You can use whatever you prefer" when we're really saying "you _should_ prefer something other than this if possible"

Reply:
Good idea. I've reworded the comment to be much stronger.